### PR TITLE
 Implement 'User Action' Functionality Similar to AnkiMobile

### LINKED ISSUE #14286: Feature Request: Implement 'User Action' Functionality Similar to AnkiMobile
In AnkiMobile (starting from version 2.0.88+), there is a feature that allows users to trigger JavaScript in the review screen. This advanced functionality is especially beneficial for those using shared decks that provide keyboard shortcuts to perform actions, such as revealing a hint or the next cloze deletion. By defining a user action, it's possible to trigger these actions with external devices, like a gamepad, or even by using gestures on the screen.

For instance, in AnkiMobile, if a card template provides a revealNext() function, users can instruct AnkiMobile to call it with the following code:

```
var userJs1 = revealNext;
```

This feature is especially handy for those who have written their own actions or are using certain shared decks/templates that provide these actions.

For more details, please refer to the [AnkiMobile documentation](https://docs.ankimobile.net/study-tools.html). Once on the page, scroll down to the "User Action 1-8" section for specifics.

### PR DESCRIPTION
**Pull request by BrayanDSO**
## Fixes
* Fixes #14286

## Approach

1. Add the preferences
2. Reverse enginner how AnkiMobiles does it and implement it here

## How Has This Been Tested?

[Screen_recording_20240216_141352.webm](https://github.com/ankidroid/Anki-Android/assets/69634269/c258c948-6f1e-40a8-8e90-4b4c4856f28c)

```text
2024-02-16 14:13:45.965 10005-10005 Reviewer                com.ichi2.anki.debug                 V  userAction1
2024-02-16 14:13:45.970 10005-10005 chromium                com.ichi2.anki.debug                 I  [INFO:CONSOLE(12)] "one", source: http://127.0.0.1:42775/ (12)
2024-02-16 14:13:46.429 10005-10097 rsdroid::logging        com.ichi2.anki.debug                 D  STDOUT: [INFO:CONSOLE(12)] "one", source: http://127.0.0.1:42775/ (12)
2024-02-16 14:13:47.532 10005-10005 AbstractFl...reDetector com.ichi2.anki.debug                 D  onFling
2024-02-16 14:13:47.532 10005-10005 Reviewer                com.ichi2.anki.debug                 V  userAction7
2024-02-16 14:13:47.535 10005-10005 chromium                com.ichi2.anki.debug                 I  [INFO:CONSOLE(14)] "seven", source: http://127.0.0.1:42775/ (14)
2024-02-16 14:13:48.430 10005-10097 rsdroid::logging        com.ichi2.anki.debug                 D  STDOUT: [INFO:CONSOLE(14)] "seven", source: http://127.0.0.1:42775/ (14)
2024-02-16 14:13:49.243 10005-10005 AbstractFl...reDetector com.ichi2.anki.debug                 D  onFling
2024-02-16 14:13:49.243 10005-10005 Reviewer                com.ichi2.anki.debug                 V  userAction9
2024-02-16 14:13:49.247 10005-10005 AbstractFl...romeClient com.ichi2.anki.debug                 I  AbstractFlashcardViewer:: onJsAlert: TypeError: userJs is not a function
```

## Learning (optional, can help others)

Adding commands needs a lot of boilerplate

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 1]
```diff
@@ -265,6 +265,16 @@
...
+    <string name="user_action" maxLength="41">User action</string>
+    <string name="user_action_1" maxLength="41">User action 1</string>
+    <string name="user_action_2" maxLength="41">User action 2</string>
+    <string name="user_action_3" maxLength="41">User action 3</string>
+    <string name="user_action_4" maxLength="41">User action 4</string>
+    <string name="user_action_5" maxLength="41">User action 5</string>
+    <string name="user_action_6" maxLength="41">User action 6</string>
+    <string name="user_action_7" maxLength="41">User action 7</string>
+    <string name="user_action_8" maxLength="41">User action 8</string>
+    <string name="user_action_9" maxLength="41">User action 9</string>
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/10-preferences.xml:277]: This seems wasteful for a number of languages, BUT:

Defining this as a plural would likely confuse translators

`User action %d` may be incorrect in a number of languages

I think this is the best possible option, but flagging for other reviewers

**Review by david-allison — APPROVED** [review]
Looks great, thanks!

File: AnkiDroid/src/main/res/values/10-preferences.xml (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/res/values/10-preferences.xml:277]: why don't we use some of the backend translations more? Just curious - this should have a backend translation already shouldn't it ?

Comment (BrayanDSO) [reply AnkiDroid/src/main/res/values/10-preferences.xml:277]: AnkiMobile has it, Anki Desktop doesn't, and our translations come from Anki Desktop

**Review by lukstbit — APPROVED** [review]
Looks good!

File: AnkiDroid/src/main/res/values/10-preferences.xml (hunk 1 above)
Comment (snowtimeglass) [reply AnkiDroid/src/main/res/values/10-preferences.xml:277]: Even AnkiMobile probably doesn't have those translations for some reason (an implication of "For advanced users"?). 


https://i18n.ankiweb.net/ja/mobile/all-resources/?search=user+action

By the way, `User action 9` is an incompatible action with AnkiMobile?

Comment (BrayanDSO) [reply AnkiDroid/src/main/res/values/10-preferences.xml:277]: > By the way, User action 9 is an incompatible action with AnkiMobile?

Huh, didn't double check. We don't need to be equivalent to AnkiMobile anyway, and extra power doesn't hurt. People can ask to Damien to add a 9th action in AnkiMobile if they want

**Comment by mikehardy** [comment]
Just wanted to note I thought this was a pretty cool bit of functionality with the reverse engineering and all 😆

**Pull request merged**